sdk/src/settings/builder.rs

```diff
@@ -323,33 +314,33 @@ impl TryFrom<ActionSettings> for Action {
 ///
 /// The reason this setting exists only for an [Actions][crate::assertions::Actions] assertion
 /// is because of its mandations and reusable fields.
-#[allow(unused)]
+#[cfg_attr(feature = "json_schema", derive(schemars::JsonSchema))]
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
-pub(crate) struct ActionsSettings {
+pub struct ActionsSettings {
     /// Whether or not to set the [Actions::all_actions_included][crate::assertions::Actions::all_actions_included]
     /// field.
     pub all_actions_included: bool,
     /// Templates to be added to the [Actions::templates][crate::assertions::Actions::templates] field.
+    #[doc(hidden)]
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub templates: Option<Vec<ActionTemplateSettings>>,
-    // TODO: should we define a new struct for "Action" too, like ActionTemplateSettings?
-    /// Actions to be added to the [Actions::actions][crate::assertions::Actions::actions] field.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub actions: Option<Vec<ActionSettings>>,
-    /// Whether to automatically generate a c2pa.created [Action][crate::assertions::Action]
-    /// assertion or error that it doesn't already exist.
+    pub(crate) templates: Option<Vec<ActionTemplateSettings>>,
+    //
+    // /// Actions to be added to the [Actions::actions][crate::assertions::Actions::actions] field.
+    // #[doc(hidden)]
+    // #[serde(skip_serializing_if = "Option::is_none")]
+    // pub(crate) actions: Option<Vec<ActionSettings>>,
+    //
+    /// Whether to automatically generate a c2pa.created [Action] assertion or error that it doesn't already exist.
     ///
     /// For more information about the mandatory conditions for a c2pa.created action assertion, see here:
     /// <https://spec.c2pa.org/specifications/specifications/2.2/specs/C2PA_Specification.html#_mandatory_presence_of_at_least_one_actions_assertion>
     pub auto_created_action: AutoActionSettings,
```

Comment:
I think we can replace auto_created_action and auto_opened_action with Builder.intent.create/open
That makes the intent clear.

Reply:
This is a good idea, but we will have to keep in mind that `BuilderIntent::Create(DigitalSourceType)` takes a value. So what it will end up looking like is:
```json
{
    "intent": {
        "type": "created",
        "source_type": "...",
    }
}
```
or
```json
{
    "intent": {
        "type": "opened"
    }
}
```